Updated to highlight that EDM tool now supports .tsv format (#5090)
MicrosoftDocs/microsoft-365-docs: microsoft-365/compliance/create-custom-sensitive-information-types-with-exact-data-match-based-classification.md

Setting up and configuring EDM-based classification involves:

1. [Saving sensitive data in .csv or .tsv format](#save-sensitive-data-in-csv-format)
2. [Define your sensitive information database schema](#define-the-schema-for-your-database-of-sensitive-information)
3. [Create a rule package](#set-up-a-rule-package)

#### Save sensitive data in .csv or .tsv format

> 1. Identify the sensitive information you want to use to identify what you want to protect. Export the data to an app, such as Microsoft Excel, and save the file in a text file. The file can be saved in .csv (comma-separated values), .tsv (tab-separated values) or pipe-separated (|) format. The .tsv format is recommended in cases where your data values may have commas included in them (e.g. street addresses).

Reviewer: ```suggestion
1. Identify the sensitive information you want to use to identify what you want to protect. Export the data to an app, such as Microsoft Excel, and save the file in a text file. The file can be saved in .csv (comma-separated values), .tsv (tab-separated values), or pipe-separated (|) format. The .tsv format is recommended in cases where your data values may have commas included in them (such as street addresses).
```

The data file can include a maximum of:
      - Up to 100 million rows of sensitive data
      - Up to 32 columns (fields) per data source
      - Up to 5 columns (fields) marked as searchable

2. Structure the sensitive data in the .csv or .tsv file such that the first row includes the names of the fields used for EDM-based classification. In your file you might have field names such as "ssn", "birthdate", "firstname", "lastname". The column header names can't include spaces or underscores. For example, the sample .csv file that we use in this article is named *PatientRecords.csv*, and its columns include *PatientID*, *MRN*, *LastName*, *FirstName*, *SSN*, and more.

> 3. Pay attention to the format of the sensitive data fields. In particular, fields that may contain commas in their content (e.g. a street address that contains the value "Seattle,WA") would be parsed as two separate fields when parsed by the EDM tool if the .csv format is selected. In order to avoid this, either use a file with the .tsv format or ensure such fields are surrounded by single or double quotes in the sensitive data table (double quotes are generally preferred since single quotes may be confused with apostrophes in names and words in certain languages). If fields with commas in them may also contain spaces, you would need to create a custom Sensitive Information Type that matches the corresponding format (e.g. a multi-word string with commas and spaces in it) to ensure the string is correctly matched when the document is scanned.

Reviewer: ```suggestion
3. Pay attention to the format of the sensitive data fields. In particular, fields that may contain commas in their content (for example, a street address that contains the value "Seattle,WA") would be parsed as two separate fields when parsed by the EDM tool if the .csv format is selected. In order to avoid this, either use a file with the .tsv format or ensure such fields are surrounded by single or double quotes in the sensitive data table (double quotes are generally preferred since single quotes may be confused with apostrophes in names and words in certain languages). If fields with commas in them also contain spaces, you would need to create a custom Sensitive Information Type that matches the corresponding format (for example, a multi-word string with commas and spaces in it) to ensure the string is correctly matched when the document is scanned.
```

- a Windows 10 or Windows Server 2016 machine with .NET version 4.6.2 for running the EDMUploadAgent
- a directory on your upload machine for the:
    -  EDMUploadAgent

>     - your sensitive item file in csv or tsv format, **PatientRecords.csv** in our examples

Reviewer: ```suggestion
    - your sensitive item file in .csv or .tsv format, **PatientRecords.csv** in our examples
```

> Before you begin this procedure, make sure that you are a member of the **EDM\_DataUploaders** security group.

> [!TIP]

> > Optionally, you can run a validation against your csv or tsv file before uploading by running:

Reviewer: ```suggestion
> Optionally, you can run a validation against your .csv or .tsv file before uploading by running:
```

4. To hash and upload the sensitive data, run the following command in Command Prompt window:

   `EdmUploadAgent.exe /UploadData /DataStoreName [DS Name] /DataFile [data file] /HashLocation [hash file location] /Schema [Schema file] /ColumnSeparator ["{Tab}"|"|"]`

   Example: **EdmUploadAgent.exe /UploadData /DataStoreName PatientRecords /DataFile C:\Edm\Hash\PatientRecords.csv /HashLocation C:\Edm\Hash /Schema edm.xml**

>    THe default format for the sensitive data file is comma-separated values. You can specify a tab-separated file by indicating the "{Tab}" option with the /ColumnSeparator parameter, or you can specify a pipe-separated file by indicating the "|" option, depending on your the format in which you exported your data. 

Reviewer: ```suggestion
   The default format for the sensitive data file is comma-separated values. You can specify a tab-separated file by indicating the "{Tab}" option with the /ColumnSeparator parameter, or you can specify a pipe-separated file by indicating the "|" option, depending on the format in which you exported your data. 
```

   - .EdmHash
   - .EdmSalt

> 2. Copy these files in a secure fashion to the computer you will use to upload your sensitive items csv or tsv file (PatientRecords) to your tenant.

Reviewer: ```suggestion
2. Copy these files in a secure fashion to the computer you will use to upload your sensitive items .csv or .tsv file (PatientRecords) to your tenant.
```

$edmuploader = $edminstallpath + 'EdmUploadAgent.exe'
$csvext = '.csv'
$schemaext = '.xml'

> \# Assuming file name is same as data store name and file is in csv format

Reviewer: ```suggestion
\# Assuming the file name is the same as the data store name and the file is in csv format
```

> \# Assuming file name is same as data store name and file is in csv format

Reviewer: ```suggestion
\# Assuming file name is same as data store name and file is in .csv format
```

$csvext = '.csv'
$edmext = '.EdmHash'
$schemaext = '.xml'

> \# Assuming file name is same as data store name and file is in csv format

Reviewer: ```suggestion
\# Assuming file name is same as data store name and file is in .csv format
```
